src/finn/transformation/fpgadataflow/convert_to_hw_layers.py

```diff
@@ -176,15 +176,21 @@ def apply(self, model):
                 thl_thres_shape = model.get_tensor_shape(thl_threshold)
                 idt = model.get_tensor_datatype(thl_input)
                 tdt = model.get_tensor_datatype(thl_threshold)
-                # skip conversion for layers with float input
-                if not idt.is_integer():
+
+                # only infer layers where input and thresholds are
```

Comment:
I think the flow could also be that the input is integer and the threshold values are still float. The transformation `RoundAndClipThresholds` can be called on MultiThreshold or Thresholding layers. (https://github.com/Xilinx/finn/blob/dev/src/finn/transformation/streamline/round_thresholds.py#L51)
Could you relax this constraint for the conversion? I think the check if the thresholds are int when the inputs are int, should be done in the code generation instead and if float thresholds are detected, the user can be asked to run `RoundAndClipThresholds` before code generation.